source/include/ota.h

```diff
@@ -177,8 +178,11 @@ typedef uint32_t OtaErr_t;
  * job has failed in some way and should be rejected.
  *
  * @param[in] eEvent An OTA update event from the OtaJobEvent_t enum.
+ *
+ * @param[in] pData Optional data related to the event.
  */
-typedef void (* OtaAppCallback_t)( OtaJobEvent_t eEvent );
+typedef void (* OtaAppCallback_t)( OtaJobEvent_t eEvent,
+                                   void * pData );
```

Comment:
Why are you using a void * here?

Reply:
We have it void ptr so that we application can apply it to any data as the data will be based on the event type.